doc/howto/network_bridge_firewalld.md

```diff
@@ -1,52 +1,120 @@
-firewall-cmd --permanent --direct --add-rule ipv4 filter FORWARD 0 -i lxdbr0 -s 10.0.0.0/24 -m comment --comment "generated by firewalld for LXD" -j ACCEPT
-firewall-cmd --permanent --direct --add-rule ipv4 nat POSTROUTING 0 -s 10.0.0.0/24 ! -d 10.0.0.0/24 -m comment --comment "generated by firewalld for LXD" -j MASQUERADE
-firewall-cmd --reload
-```
-To check the rules are taken into account by firewalld:
+If UFW has a rule to drop all unrecognised traffic, it blocks the traffic from and to the LXD bridge.
+In this case, you must add rules to allow traffic from and to the bridge.
+
+To do so, run the following commands:
+
+    sudo ufw allow in on <network_bridge>
+    sudo ufw route allow in on <network_bridge>
+
+For example:
+
+    sudo ufw allow in on lxdbr0
+    sudo ufw route allow in on lxdbr0
+
+## Prevent rule conflicts
+
+Linux firewalls are based on netfilter.
+LXD uses the same subsystem, which can lead to conflicts if the defined rules overlap.
+For example, the firewall might erase LXD rules if it is started after the LXD daemon, which might interrupt network connectivity to the instance.
+
+In this case, you should directly configure the rules defined by LXD in the firewall, and disable them in LXD.
+
+### xtables vs. nftables
+
+There are different userspace commands to add rules to netfilter: xtables (iptables for IPv4 and ip6tables for IPv6) and nftables.
+
+Xtables provides an ordered list of rules, which might cause issues if multiple systems add and remove entries from the list.
+Nftables adds the ability to separate rules into namespaces, which helps to separate rules from different applications.
+However, if a packet is blocked in one namespace, it is not possible for another namespace to allow it.
+Therefore, rules in one namespace can still affect rules in another namespace, and firewall applications can still impact LXD network functionality.
+
+If your system supports and uses nftables, LXD detects this and switches to nftables mode.
+In this mode, LXD adds its rules into the nftables, using its own nftables namespace.
+
+### Disable LXD's firewall rules
+
+Run the following commands to prevent LXD from setting firewall rules for a specific network bridge (for example, `lxdbr0`):
+
+    lxc network set <network_bridge> ipv4.nat false
+    lxc network set <network_bridge> ipv6.nat false
+    lxc network set <network_bridge> ipv6.firewall false
+    lxc network set <network_bridge> ipv4.firewall false
+
+### Firewalld: Add LXD firewall rules
+
+To configure the LXD firewall rules in firewalld, you must add input, output, forward and postrouting rules for the network bridge (for example, `lxdbr0`) and the associated IP range (for example, `192.0.2.0/24`):
+
+    sudo firewall-cmd --permanent --direct --add-rule ipv4 filter INPUT 0 -i <network_bridge> -s <IP_range> -m comment --comment "generated by firewalld for LXD" -j ACCEPT
```

Comment:
So I think we can get rid of this section, firstly because adding lxdbr0 to the trusted zone (above) gets connectivity working, and also because there appears to be an issue in firewalld when using these `--direct` rules that they are added to the wrong chain and so are not taking effect on recent versions of firewalld. Also there is another issue were if you have iptables legacy and nftables installed (but not the nftables iptables shim), then the direct rules get added to iptables, and the rest of the rules get added to nftables. So I think it would be good to steer clear of these commands altogether.